examples/chat-example/src/main.rs

```diff
@@ -46,34 +46,49 @@
 //! event and remove the expired peer from the list of known peers.
 
 use async_std::io;
-use futures::{prelude::*, select};
+use futures::{future::Either, prelude::*, select};
 use libp2p::{
-    gossipsub, identity, mdns,
+    core::{muxing::StreamMuxerBox, transport::OrTransport, upgrade},
+    gossipsub, identity, mdns, mplex, noise, quic,
     swarm::NetworkBehaviour,
     swarm::{SwarmBuilder, SwarmEvent},
-    PeerId,
+    tcp, PeerId, Transport,
 };
 use std::collections::hash_map::DefaultHasher;
 use std::error::Error;
 use std::hash::{Hash, Hasher};
 use std::time::Duration;
 
+// We create a custom network behaviour that combines Gossipsub and Mdns.
+#[derive(NetworkBehaviour)]
+struct MyBehaviour {
+    gossipsub: gossipsub::Behaviour,
+    mdns: mdns::async_io::Behaviour,
+}
```

Comment:
Why is this here if we don't use it?

Reply:
`MyBehaviour` is used below, I just moved it from the `main`.

Reply:
Sorry, my bad.